pkg/storage/ingest/reader.go

```diff
@@ -1016,7 +1062,8 @@ func (r *concurrentFetchers) runFetchers(ctx context.Context, startOffset int64)
 				}
 				continue
 			}
-			bytesPerRecord = estimateBytesPerRecord(bytesPerRecord, result.fetchedBytes, len(result.Records))
+			nextFetch.bytesPerRecord = estimateBytesPerRecord(nextFetch.bytesPerRecord, result.fetchedBytes, len(result.Records))
+			nextFetch = nextFetch.trimIfTooBig()
```

Comment:
good call. I chose `UpdateBytesPerRecord` because `expectedBytes()` aren't exported, but also `UpdateMaxBytes` sounded strange (isn't MaxBytes a pure function how are we updating it?)